tools/sanity_checks.py

```diff
@@ -361,52 +364,46 @@ def check_new_release(self, name: str, builddir: str = '_build', deps=None, prog
         msys_packages = ci.get('msys_packages', [])
         alpine_packages = ci.get('alpine_packages', [])
         meson_env = os.environ.copy()
-        if debian_packages and is_debianlike():
+        def install_packages(cmd, packages):
             if is_ci():
-                subprocess.check_call(['sudo', 'apt-get', '-y', 'install', '--no-install-recommends'] + debian_packages)
+                with ci_group('install packages'):
+                    subprocess.check_call(cmd + packages)
             else:
-                s = ', '.join(debian_packages)
+                s = ', '.join(packages)
                 print(f'The following packages could be required: {s}')
+        if debian_packages and is_debianlike():
+            install_packages(['sudo', 'apt-get', '-y', 'install', '--no-install-recommends'], debian_packages)
         elif brew_packages and is_macos():
+            install_packages(['brew', 'install', '--quiet'], brew_packages)
             if is_ci():
-                subprocess.check_call(['brew', 'install'] + brew_packages)
-            else:
-                s = ', '.join(brew_packages)
-                print(f'The following packages could be required: {s}')
+                # Ensure binaries from keg-only formulas are available (e.g. bison).
+                out = subprocess.check_output(['brew', '--prefix'] + brew_packages)
+                for prefix in out.decode().split('\n'):
+                    bindir = Path(prefix) / 'bin'
+                    if bindir.exists():
+                        meson_env['PATH'] = str(bindir) + ':' + meson_env['PATH']
```

Comment:
This means that `brew` installed binaries will silently override system ones.  Is that something we truly want? In theory this could be used to hijack commands, but in practice it would mean that `brew` is already compromised so they can get in via other means.

Reply:
Yeah, formulas are ruby code, no?